pkg/azclient/metrics/metrics.go

```diff
@@ -0,0 +1,122 @@
+package metrics
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/metric"
+	"k8s.io/klog/v2"
+)
+
+var (
+	defaultMetrics *Metrics
+)
+
+func init() {
+	var err error
+	defaultMetrics, err = New()
+	if err != nil {
+		klog.Fatalf("create default metrics: %v", err)
+	}
+}
+
+func Default() *Metrics {
+	return defaultMetrics
+}
+
+type Metrics struct {
+	meter               metric.Meter
+	apiLatency          metric.Float64Histogram
+	apiErrorCount       metric.Int64Counter
+	apiRateLimitedCount metric.Int64Counter
+	apiThrottledCount   metric.Int64Counter
+}
+
+func New() (*Metrics, error) {
+	meter := otel.Meter("cloud-provider-azclient")
```

Comment:
I'm wondering if we can put these code into init func so that the metrics will be registered when the package is imported.